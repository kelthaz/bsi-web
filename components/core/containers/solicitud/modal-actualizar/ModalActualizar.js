import { useRouter } from 'next/router';
import { useDispatch, useSelector } from 'react-redux';
import { useEffect, useState } from 'react';
import Modal from '../../../../shared/modal/Modal';
import { resetChangePage, onHandleSubmit } from '../../../../../redux/actions/formulario';

const ModalActualizar = () => {
  const { push } = useRouter();
  const dispatch = useDispatch();
  const { routePage, showModal, isValid } = useSelector((state) => state.formulario);
  const [openModal, setOpenModal] = useState(false);

  const handleModalFirstOption = () => {
    setOpenModal(false);
    if (isValid) {
      dispatch(onHandleSubmit(true));
    } else {
      dispatch(resetChangePage());
    }
  };

  const handleModalSecondOption = async () => {
    setOpenModal(false);
    dispatch(resetChangePage());
    await push(routePage);
  };

  useEffect(() => {
    if (showModal) {
      setOpenModal(true);
    }
  }, [showModal]);

  return (
    <Modal openModal={openModal} setOpenModal={setOpenModal} onClose={() => dispatch(resetChangePage())}>
      <div className="modal-alerta-container">
        <h4 className="color-blue-storm">
          {isValid
            ? 'Se han detectado cambios en formulario...'
            : 'Se han detectado cambios con errores en el formulario...'}
        </h4>
        <p className="dark-gray body2">
          {isValid
            ? 'Desea guardar los cambios o descartarlos.'
            : 'Desea permanecer y corregir los errores o desea descartar los cambios.'}
        </p>
        <div className="flex-row-center-config">
          <button type="button" className="btn-medium-secondary mr-2" onClick={handleModalFirstOption}>
            {isValid ? 'Sí, guardar' : 'Sí, corregir'}
          </button>

          <button type="button" className="btn-medium ml-2" onClick={handleModalSecondOption}>
            No, descartar
          </button>
        </div>
      </div>
    </Modal>
  );
};

export default ModalActualizar;
