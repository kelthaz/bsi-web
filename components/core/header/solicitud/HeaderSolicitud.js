import React from 'react';

const HeaderSolicitud = () => {
  return <div></div>;
};

export default HeaderSolicitud;
