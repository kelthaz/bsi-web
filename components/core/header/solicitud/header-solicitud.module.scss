@import '../../../../styles/scss/modules/var';
@import '../../../../styles/scss/modules/mixins';

.container-header {
  align-items: center;
  border-bottom: 1px solid $blue-shadow;
  display: flex;
  flex-direction: row;
  justify-content: space-between;
  padding: 16px 32px;

  @include media('sm') {
    padding: 14px 16px;
  }
}
