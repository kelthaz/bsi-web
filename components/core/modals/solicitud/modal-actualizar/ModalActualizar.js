import { useRouter } from 'next/router';
import { useSelector } from 'react-redux';
import Modal from '../../../../shared/modal/Modal';

const ModalActualizar = ({ openModal, setOpenModal, formulario }) => {
  const { push } = useRouter();
  const { routePage } = useSelector((state) => state.formulario);

  const handleModalFirstOption = () => {
    if (formulario.isValid) {
      formulario.handleSubmit();
      push('/solicitud/[tab]/[step]', routePage);
    }
    setOpenModal(false);
  };

  const handleModalSecondOption = () => {
    push('/solicitud/[tab]/[step]', routePage);
    setOpenModal(false);
  };

  return (
    <Modal openModal={openModal} setOpenModal={setOpenModal}>
      <div className="modal-alerta-container">
        <h4 className="color-blue-storm">
          {formulario.isValid
            ? 'Se han detectado cambios en formulario...'
            : 'Se han detectado cambios con errores en el formulario...'}
        </h4>
        <p className="dark-gray body2">
          {formulario.isValid
            ? 'Desea guardar los cambios o descartarlos.'
            : 'Desea permanecer y corregir los errores o desea descartar los cambios.'}
        </p>
        <div className="flex-row-center-config">
          <button type="button" className="btn-medium-secondary mr-2" onClick={handleModalFirstOption}>
            {formulario.isValid ? 'Sí, guardar' : 'Sí, corregir'}
          </button>

          <button type="button" className="btn-medium ml-2" onClick={handleModalSecondOption}>
            No, descartar
          </button>
        </div>
      </div>
    </Modal>
  );
};

export default ModalActualizar;
