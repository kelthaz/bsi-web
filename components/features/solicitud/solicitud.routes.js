import cargaDocumentosRoutes from './carga-documentos/carga-documentos.routes';
import datosEmpresaRoutes from './datos-empresa/datos-empresa.routes';
import datosPersonalesRoutes from './datos-personales/datos-personales.routes';
import documentacionRoutes from './documentacion/documentacion.routes';
import ofertaRoutes from './oferta/oferta.routes';

const solicitudRoutes = [
  ...datosPersonalesRoutes,
  // ...datosEmpresaRoutes,
  // ...documentacionRoutes,
  // ...cargaDocumentosRoutes,
  // ...ofertaRoutes
].map((route) => ({ ...route, feature: 'solicitud' }));

export default solicitudRoutes;
