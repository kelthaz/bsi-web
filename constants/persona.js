export const MORAL = 'MORAL';
export const FISICA = 'FISICA';
