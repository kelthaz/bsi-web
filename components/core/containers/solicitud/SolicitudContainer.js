import { useRouter } from 'next/router';
import PropTypes from 'prop-types';
import { useEffect } from 'react';
import { useSelector } from 'react-redux';
import { DATO_PERSONA } from '../../../../constants/formularios';
import { SIMULADOR_ROUTE } from '../../../../constants/routes/publico/publico';
import solicitudRoutes from '../../../features/solicitud/solicitud.routes';
import Step from '../../../shared/step/Step';
import TabInformativo from '../../../shared/tab-informativo/TabInformativo';
import SvgPatronesSolicitud from '../../../svgs/SvgPatronesSolicitud';
import HeaderSolicitud from '../../header/solicitud/HeaderSolicitud';
import ModalActualizar from '../../modals/solicitud/modal-actualizar/ModalActualizar';

const SolicitudContainer = ({ pageComponent, servicesData }) => {
  const { component, data } = pageComponent;
  const [componentPFAE, componentPM] = component;
  const { formulario, paso, step, tab, tipoPersona } = data;
  const { sincronizado, currentStep, tipoPersonaRedux } = useSelector((state) => state.solicitud);
  const { replace } = useRouter();

  let Component = <></>;

  if (tipoPersona === '' && componentPM) {
    Component = tipoPersonaRedux === 'MORAL' ? componentPM : componentPFAE;
  } else {
    Component = componentPFAE;
  }

  const componentsFormulario = solicitudRoutes
    .filter(({ data: dataComponent }) => dataComponent.formulario === formulario)
    .map(({ data: dataComponent, route }) => ({
      step: dataComponent.step,
      paso: dataComponent.paso,
      tipoPersona: dataComponent.tipoPersona,
      action: route,
    }));

  const steps = componentsFormulario.filter(
    (dataComponent) =>
      Number.isInteger(dataComponent.step) &&
      (dataComponent.tipoPersona === '' || dataComponent.tipoPersona === tipoPersonaRedux)
  );

  const tabs = [
    { path: 'datos-personales', label: 'Datos personales' },
    { path: 'datos-empresa', label: 'Datos de tu empresa' },
    { path: 'oferta', label: 'Oferta' },
    { path: 'documentacion', label: 'DocumentaciÃ³n' },
  ];

  useEffect(() => {
    if (!sincronizado && formulario === DATO_PERSONA) {
      replace(SIMULADOR_ROUTE);
    }
  }, []);

  return (
    <>
      <HeaderSolicitud />
      <ModalActualizar />

      {!!tab && (
        <TabInformativo
          tabs={tabs}
          currentTab={tab}
          currentStep={paso}
          valipStep={currentStep.paso}
          steps={componentsFormulario}
        />
      )}

      {!!step && <Step currentStep={step} valipStep={currentStep.valipStep} steps={steps} />}
      <Component {...servicesData} validate={currentStep.paso === paso} />
      <SvgPatronesSolicitud className="fixed-left-bottom" />
    </>
  );
};

SolicitudContainer.propTypes = {
  pageComponent: PropTypes.any.isRequired,
  servicesData: PropTypes.any.isRequired,
};

export default SolicitudContainer;
