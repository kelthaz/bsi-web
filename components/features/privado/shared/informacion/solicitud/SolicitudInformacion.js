import React from 'react';
import PropTypes from 'prop-types';
import { MORAL } from '../../../../../../constants/persona';

const SolicitudInformacion = ({ informacionSolicitud, tipoPersona }) => {
  const labels = {
    usoCredito: 'Destino del crédito',
    descripcionCredito: 'Razón social',
    domicilioFiscal: 'Domicilio fiscal',
    domicilioComercial: 'Domicilio comercial',
    nombreCompletoRecibe: 'Nombre de quien podría recibir token',
    celularRecibe: 'Teléfono de quien podría recibir token',
    telefonoEmpresa: 'Teléfono del negocio',
    curp: tipoPersona === MORAL ? 'CURP' : 'CURP del representante legal',
    estadoCivil: 'Estado civil del solicitante',
    nombreConyuge: 'Nombre de la Pareja',
    numeroEmpleados: 'Número de empleados',

    montoSolicitado: 'Crédito solicitado',
    montoAprobado: 'Crédito aprobado',

    tasaOrdinaria: 'Tasa ordinaria',
    tasaMoratoria: 'Tasa moratoria',
    comisionApertura: 'Comisión por apertura',
    plazo: 'Plazo del crédito',
    periodicidad: 'Esquema de pago',
    pago: 'Pagos',
    cat: 'CAT',
    tieneCuentaBancoppel: 'Tiene cuenta BanCoppel',
    clabe: 'Número de cuenta BanCoppel',
  };

  if (tipoPersona === MORAL) {
    delete labels.estadoCivil;
    delete labels.nombreConyuge;
  }

  const itemsPerfil = Object.keys(labels);

  return (
    <>
      {itemsPerfil.map((item) => (
        <div key={item} className="row pb-3">
          <div className="col-6">
            <div className="body2">{labels[item]}</div>
          </div>
          <div className="col-6">
            <div className="color-blue-storm body2">{informacionSolicitud[item]}</div>
          </div>
        </div>
      ))}
    </>
  );
};

SolicitudInformacion.propTypes = {
  informacionSolicitud: PropTypes.object.isRequired,
  tipoPersona: PropTypes.string.isRequired,
};

export default SolicitudInformacion;
