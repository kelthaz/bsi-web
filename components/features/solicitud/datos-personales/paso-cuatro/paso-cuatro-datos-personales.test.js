import '@testing-library/jest-dom';
import React from 'react';
import { mount } from 'enzyme';
import { Provider } from 'react-redux';
import { useRouter } from 'next/router';
import { act } from 'react-dom/test-utils';
import storeTest from '../../../../../redux/storeTest';
import PasoCuatroDatosPersonales from './PasoCuatroDatosPersonales';
import { PASO_CINCO_DATOS_PERSONA_ROUTE } from '../../../../../constants/routes/solicitud/persona';
import EmailageRepositorio from '../../../../../services/solicitud/emailage.repositorio';
import ModalActualizar from '../../../../core/containers/solicitud/modal-actualizar/ModalActualizar';

jest.mock('../../../../../services/solicitud/emailage.repositorio');

describe('Pruebas en el componente PasoCuatroDatosPersonales', () => {
  const push = jest.fn();
  const beforePopState = jest.fn();
  useRouter.mockImplementation(() => ({ push, beforePopState }));

  const emailage = {
    email: 'jesus.gomezo@udea.edu.com',
    fraudRisk: '500 Moderate',
    emailExists: 'Yes',
    domainExists: 'Yes',
  };

  beforeEach(() => {
    jest.clearAllMocks();
  });

  test('Debe de mostrarse correctamente el componente', async () => {
    const store = storeTest();
    const wrapper = mount(
      <Provider store={store}>
        <PasoCuatroDatosPersonales validate />
        <ModalActualizar />
      </Provider>
    );
    expect(wrapper).toMatchSnapshot();
    wrapper.unmount();
  });

  test('Debe de guardar en el reducer el nuevo paso, celular, correo y enviarlo al siguiente paso, como persona Fisica', async () => {
    // arrange
    const store = storeTest();
    const {
      solicitud: { currentStep },
    } = store.getState();
    const wrapper = mount(
      <Provider store={store}>
        <PasoCuatroDatosPersonales validate />
        <ModalActualizar />
      </Provider>
    );

    await EmailageRepositorio.postEmailScore.mockResolvedValue({ data: emailage });

    const form = wrapper.find('form');
    const celular = wrapper.find('input').find({ name: 'celular' });
    const correo = wrapper.find('input').find({ name: 'correo' });
    // act
    await act(async () => {
      celular.simulate('change', {
        target: { value: '33-3333-3333', selectionStart: 0, setSelectionRange: jest.fn() },
      });
    });

    await act(async () => {
      correo.simulate('change', {
        target: { value: 'jesus.gomezo@udea.edu.com', selectionStart: 0, setSelectionRange: jest.fn() },
      });
    });

    await act(async () => {
      form.simulate('submit');
    });

    const {
      solicitud: { datosPersonales },
    } = store.getState();
    // // assert
    expect(datosPersonales.celular).toEqual(wrapper.find('input').find({ name: 'celular' }).prop('value'));
    expect(datosPersonales.correo).toEqual(wrapper.find('input').find({ name: 'correo' }).prop('value'));

    expect(store.getState().solicitud.currentStep.paso).toBe(currentStep.paso + 1);
    expect(store.getState().solicitud.currentStep.valipStep).toBe(currentStep.valipStep + 1);
    expect(push).toHaveBeenCalledTimes(1);
    expect(push).toHaveBeenCalledWith(PASO_CINCO_DATOS_PERSONA_ROUTE);
    wrapper.unmount();
  });

  test('Debe de no guardar nombreEmpresa, sector, giro, descripcionEmpresa y no enviarlo al siguiente paso, como persona Moral', async () => {
    // arrange
    const store = storeTest();
    const {
      solicitud: { currentStep },
    } = store.getState();
    const wrapper = mount(
      <Provider store={store}>
        <PasoCuatroDatosPersonales validate />
        <ModalActualizar />
      </Provider>
    );

    await EmailageRepositorio.postEmailScore.mockRejectedValue({
      response: {
        data: { message: ['Ocurrio un error al consultar el correo'] },
      },
    });

    const form = wrapper.find('form');
    const celular = wrapper.find('input').find({ name: 'celular' });
    const correo = wrapper.find('input').find({ name: 'correo' });
    // act
    await act(async () => {
      celular.simulate('change', {
        target: { value: '33-3333-3333', selectionStart: 0, setSelectionRange: jest.fn() },
      });
    });

    await act(async () => {
      correo.simulate('change', {
        target: { value: 'jesus.gomezo@udea.edu.com', selectionStart: 0, setSelectionRange: jest.fn() },
      });
    });

    await act(async () => {
      form.simulate('submit');
    });

    const {
      solicitud: { datosPersonales },
    } = store.getState();
    // // assert
    expect(datosPersonales.celular).not.toEqual(wrapper.find('input').find({ name: 'celular' }).prop('value'));
    expect(datosPersonales.correo).not.toEqual(wrapper.find('input').find({ name: 'correo' }).prop('value'));

    expect(store.getState().solicitud.currentStep.paso).toBe(currentStep.paso);
    expect(store.getState().solicitud.currentStep.valipStep).toBe(currentStep.valipStep);
    expect(push).toHaveBeenCalledTimes(0);
    wrapper.unmount();
  });

  test('Debe de mantener el paso anterior en el reducer y actualizar los datos que se cambiaron y enviarlo al siguiente paso', async () => {
    // arrange
    const store = storeTest();
    const {
      solicitud: { currentStep },
    } = store.getState();
    const wrapper = mount(
      <Provider store={store}>
        <PasoCuatroDatosPersonales validate={false} />
        <ModalActualizar />
      </Provider>
    );

    await EmailageRepositorio.postEmailScore.mockResolvedValue({ data: emailage });

    const form = wrapper.find('form');
    const celular = wrapper.find('input').find({ name: 'celular' });
    const correo = wrapper.find('input').find({ name: 'correo' });
    // act
    await act(async () => {
      celular.simulate('change', {
        target: { value: '33-3333-3333', selectionStart: 0, setSelectionRange: jest.fn() },
      });
    });

    await act(async () => {
      correo.simulate('change', {
        target: { value: 'jesus.gomezo@udea.edu.com', selectionStart: 0, setSelectionRange: jest.fn() },
      });
    });

    await act(async () => {
      form.simulate('submit');
    });

    wrapper.update();

    await act(async () => {
      wrapper.find('.btn-medium-secondary').simulate('click');
    });

    const {
      solicitud: { datosPersonales },
    } = store.getState();

    // assert//
    expect(datosPersonales.celular).toEqual(wrapper.find('input').find({ name: 'celular' }).prop('value'));
    expect(datosPersonales.correo).toEqual(wrapper.find('input').find({ name: 'correo' }).prop('value'));
    expect(store.getState().solicitud.currentStep.paso).toBe(currentStep.paso);
    expect(store.getState().solicitud.currentStep.valipStep).toBe(currentStep.valipStep);
    expect(push).toHaveBeenCalledTimes(1);
    expect(push).toHaveBeenCalledWith(PASO_CINCO_DATOS_PERSONA_ROUTE);
    wrapper.unmount();
  });
});
