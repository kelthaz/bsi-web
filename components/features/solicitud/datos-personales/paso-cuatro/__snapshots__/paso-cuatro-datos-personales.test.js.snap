// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el componente PasoCuatroDatosPersonales Debe de mostrarse correctamente el componente 1`] = `
<Provider
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
      Symbol(observable): [Function],
    }
  }
>
  <PasoCuatroDatosPersonales
    validate={true}
  >
    <div
      className="contedor-fixed-tab"
    >
      <div
        className="contedor-solicitud"
      >
        <div
          className="container p-0"
        >
          <form
            noValidate={true}
            onSubmit={[Function]}
          >
            <h2
              className="color-blue-storm text-overflow"
            >
              ¡Perfecto, 
              !
            </h2>
            <p
              className="color-dark-gray sub position-relative"
            >
              <span>
                Ahora, ¿cuál es tu número de teléfono celular y correo electrónico?
              </span>
              <Tooltip
                message="Nos servirá para llamarte o enviarte mensajes de texto para dar seguimiento a tu solicitud del crédito."
                position="bottom"
              >
                <span
                  className="tooltip"
                >
                   
                  <SvgInformation>
                    <svg
                      fill="none"
                      height={18}
                      viewBox="0 0 18 18"
                      width={18}
                    >
                      <path
                        d="M11.75 13.813a.687.687 0 00-.688-.688h-.687v-4.64a.86.86 0 00-.86-.86h-1.89a.687.687 0 100 1.375h.688v4.125h-.688a.687.687 0 100 1.375h3.438c.38 0 .687-.308.687-.688zM10.375 4.875a1.375 1.375 0 11-2.75 0 1.375 1.375 0 012.75 0z"
                        fill="#338CBF"
                      />
                      <path
                        clipRule="evenodd"
                        d="M.75 9a8.25 8.25 0 1116.5 0A8.25 8.25 0 01.75 9zM9 2.125a6.875 6.875 0 100 13.75 6.875 6.875 0 000-13.75z"
                        fill="#338CBF"
                        fillRule="evenodd"
                      />
                    </svg>
                  </SvgInformation>
                </span>
                <span
                  className="tooltiptext position-bottom"
                >
                  Nos servirá para llamarte o enviarte mensajes de texto para dar seguimiento a tu solicitud del crédito.
                </span>
              </Tooltip>
            </p>
            <div
              className="row no-gutters"
            >
              <div
                className="col-lg-4 col-md-4 col-sm-12 col-xs-12"
              >
                <p
                  className="input color-gray"
                >
                  <span>
                    Mi número es
                  </span>
                </p>
              </div>
              <div
                className="col-lg-8 col-md-8 col-sm-12 col-xs-12 pr-lg-2 pr-md-2"
              >
                <TextField
                  capitalize={false}
                  disabled={false}
                  error=""
                  format="phone"
                  initialTouched={false}
                  initialValue=""
                  inverted={false}
                  label="55-9999-9999"
                  maxlength={12}
                  name="celular"
                  optional={false}
                  paste={true}
                  readonly={false}
                  setError={[Function]}
                  setTouched={[Function]}
                  setValue={[Function]}
                  size="big"
                  touched={false}
                  type="tel"
                  value=""
                >
                  <div
                    className="group"
                  >
                    <input
                      autoComplete="off"
                      className="input-big  "
                      disabled={false}
                      id="celular"
                      maxLength={12}
                      name="celular"
                      onBeforeInput={[Function]}
                      onBlur={[Function]}
                      onChange={[Function]}
                      onKeyDown={[Function]}
                      onPaste={[Function]}
                      placeholder="55-9999-9999"
                      readOnly={false}
                      tabIndex="0"
                      type="tel"
                      value=""
                    />
                    <span>
                       
                    </span>
                  </div>
                </TextField>
              </div>
            </div>
            <div
              className="row no-gutters"
            >
              <div
                className="col-12"
              >
                <p
                  className="input color-gray"
                >
                  y mi correo electrónico es
                </p>
              </div>
            </div>
            <div
              className="row no-gutters"
            >
              <div
                className="col-12"
              >
                <TextField
                  capitalize={false}
                  disabled={false}
                  error=""
                  format="email"
                  initialTouched={false}
                  initialValue=""
                  inverted={false}
                  label="ejemplo@mail.com"
                  maxlength={100}
                  name="correo"
                  optional={false}
                  paste={true}
                  readonly={false}
                  setError={[Function]}
                  setTouched={[Function]}
                  setValue={[Function]}
                  size="big"
                  touched={false}
                  type="email"
                  value=""
                >
                  <div
                    className="group"
                  >
                    <input
                      autoComplete="off"
                      className="input-big  "
                      disabled={false}
                      id="correo"
                      maxLength={100}
                      name="correo"
                      onBeforeInput={[Function]}
                      onBlur={[Function]}
                      onChange={[Function]}
                      onKeyDown={[Function]}
                      onPaste={[Function]}
                      placeholder="ejemplo@mail.com"
                      readOnly={false}
                      tabIndex="0"
                      type="email"
                      value=""
                    />
                    <span>
                       
                    </span>
                  </div>
                </TextField>
              </div>
            </div>
            <div
              className="flex-column-center-config"
            >
              <button
                aria-label="Avanzar"
                className="cicle-button-blue my-3"
                disabled={true}
                type="submit"
              />
            </div>
          </form>
        </div>
      </div>
    </div>
  </PasoCuatroDatosPersonales>
  <ModalActualizar>
    <Modal
      closeable={true}
      onClose={[Function]}
      openModal={false}
      setOpenModal={[Function]}
    />
  </ModalActualizar>
</Provider>
`;
